inc/classes/admin/settings/class-settings.php

```diff
@@ -0,0 +1,470 @@
+	 * @param array $input Array of values submitted by the form.
+	 * @return array
+	 */
+	public function sanitize_callback( $input ) {
+		$input['do_beta'] = ! empty( $input['do_beta'] ) ? 1 : 0;
+
+		$input['cache_logged_user'] = ! empty( $input['cache_logged_user'] ) ? 1 : 0;
+
+		$input['cache_mobile']            = ! empty( $input['cache_mobile'] ) ? 1 : 0;
+		$input['do_caching_mobile_files'] = ! empty( $input['do_caching_mobile_files'] ) ? 1 : 0;
+
+		$input['minify_google_fonts'] = ! empty( $input['minify_google_fonts'] ) ? 1 : 0;
+		$input['minify_html']         = ! empty( $input['minify_html'] ) ? 1 : 0;
+
+		// Option : Minification CSS & JS.
+		$input['minify_css'] = ! empty( $input['minify_css'] ) ? 1 : 0;
+		$input['minify_js']  = ! empty( $input['minify_js'] ) ? 1 : 0;
+
+		$input['minify_concatenate_css'] = ! empty( $input['minify_concatenate_css'] ) ? 1 : 0;
+		$input['minify_concatenate_js']  = ! empty( $input['minify_concatenate_js'] ) ? 1 : 0;
+
+		$input['defer_all_js']      = ! empty( $input['defer_all_js'] ) ? 1 : 0;
+		$input['defer_all_js_safe'] = ! empty( $input['defer_all_js_safe'] ) ? 1 : 0;
+
+		// Force mobile cache & specific mobile cache if one of the mobile plugins is active.
+		if ( rocket_is_mobile_plugin_active() ) {
+			$input['cache_mobile']            = 1;
+			$input['do_caching_mobile_files'] = 1;
+		}
+
+		$input['embeds'] = ! empty( $input['embeds'] ) ? 1 : 0;
+		$input['emoji']  = ! empty( $input['emoji'] ) ? 1 : 0;
+
+		$input['lazyload']         = ! empty( $input['lazyload'] ) ? 1 : 0;
+		$input['lazyload_iframes'] = ! empty( $input['lazyload_iframes'] ) ? 1 : 0;
+		$input['lazyload_youtube'] = ! empty( $input['lazyload_youtube'] ) ? 1 : 0;
+
+		// If iframes lazyload is not checked, uncheck youtube thumbnail option too.
+		if ( empty( $input['lazyload_iframes'] ) ) {
+			$input['lazyload_youtube'] = 0;
+		}
+
+		// Option : Purge interval.
+		$input['purge_cron_interval'] = isset( $input['purge_cron_interval'] ) ? (int) $input['purge_cron_interval'] : $this->options->get( 'purge_cron_interval' );
+		$input['purge_cron_unit']     = isset( $input['purge_cron_unit'] ) ? $input['purge_cron_unit'] : $this->options->get( 'purge_cron_unit' );
+
+		// Force a minimum 10 minutes value for the purge interval.
+		if ( $input['purge_cron_interval'] < 10 && 'MINUTE_IN_SECONDS' === $input['purge_cron_unit'] ) {
+			$input['purge_cron_interval'] = 10;
+		}
+
+		// Option : Remove query strings.
+		$input['remove_query_strings'] = ! empty( $input['remove_query_strings'] ) ? 1 : 0;
+
+		// Option : Prefetch DNS requests.
+		if ( ! empty( $input['dns_prefetch'] ) ) {
+			if ( ! is_array( $input['dns_prefetch'] ) ) {
+				$input['dns_prefetch'] = explode( "\n", $input['dns_prefetch'] );
+			}
+			$input['dns_prefetch'] = array_map( 'trim', $input['dns_prefetch'] );
+			$input['dns_prefetch'] = array_map( 'esc_url', $input['dns_prefetch'] );
+			$input['dns_prefetch'] = (array) array_filter( $input['dns_prefetch'] );
```

Comment:
Array cast is unnecessary, `array_filter()` always returns an array. Same goes for the next ones.